test/e2e/operator_test.go

```diff
@@ -2057,6 +2059,226 @@ func TestNetworkLoadBalancer(t *testing.T) {
+	route.Spec.Host = fmt.Sprintf("%s-%s.%s", route.Name, route.Namespace, ic.Spec.Domain)
+	if err := kclient.Create(context.TODO(), route); err != nil {
+		t.Fatalf("failed to create route %s/%s: %v", route.Namespace, route.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), route); err != nil {
+			t.Fatalf("failed to delete route %s/%s: %v", route.Namespace, route.Name, err)
+		}
+	}()
+
+	// Wait for the load balancer and DNS to be ready.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Verify that the ingresscontroller has the expected annotation.
+	lbService := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get LoadBalancer service: %v", err)
+	}
+
+	const key = "service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout"
+	expected := "3"
+	if v, ok := lbService.Annotations[key]; !ok {
+		t.Fatalf("missing expected %s=%s annotation: %+v", key, expected, lbService.Annotations)
+	} else if v != expected {
+		t.Fatalf("expected %s=%s, found %s=%s", key, expected, key, v)
+	}
+
+	// Make sure we can resolve the route's host name.  It may take some
+	// time for the ingresscontroller's wildcard DNS record to propagate.
+	if err := wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
+		_, err := net.LookupIP(route.Spec.Host)
+		if err != nil {
+			t.Log(err)
+			return false, nil
+		}
+
+		return true, nil
+	}); err != nil {
+		t.Errorf("failed to observe expected condition: %v", err)
+	}
+
+	// Open a connection to the route, send a request, and verify that the
+	// connection times out after ~10 seconds.
+	request, err := http.NewRequest("GET", "http://"+route.Spec.Host, nil)
+	if err != nil {
+		t.Fatalf("failed to create HTTP request: %v", err)
+	}
+
+	client := &http.Client{}
+
+	if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		start := time.Now()
+		response, err := client.Do(request)
+		if err != nil {
+			elapsed := time.Now().Sub(start)
+
+			// Ignore errors other than EOF.
+			if !errors.Is(err, io.EOF) {
+				t.Logf("got unexpected error after elapsed time %v: %v", elapsed, err)
+				return false, nil
+			}
+
+			// Allow up to 15 seconds.  This is well above the
+			// configured connection idle timeout, but it is also
+			// well below the default idle timeout, so it should be
+			// unlikely to cause false negatives or false positives.
+			if elapsed.Seconds() > float64(15) {
+				t.Logf("got expected EOF error after unexpectedly long elapsed time %v", elapsed)
+				return false, nil
+			}
+
+			t.Logf("got expected EOF after elapsed time %v", elapsed)
+			return true, nil
+		} else {
+			defer response.Body.Close()
```

Comment:
Do you need to close the body before you exit the polling loop?  It's just a test, but it still looks like it's in the wrong place to me.

Reply:
We don't need to close the body if the client got an error.  The deferred call to `response.Body.Close()` will run when the polling loop function returns (note the `func()` in `wait.PollImmediate(..., ..., func() (bool, error) {`), so I think this logic is correct and will close the body every time the polling loop runs and makes a successful client request.